src/goto-instrument/loop_utils.cpp

```diff
@@ -107,3 +108,41 @@ void get_modifies(
     }
   }
 }
+
+// TODO handle aliasing at all
+void get_uses(
+  const local_may_aliast &local_may_alias,
+  const loopt &loop,
+  usest &uses)
+{
+  for(loopt::const_iterator
+      i_it=loop.begin(); i_it!=loop.end(); i_it++)
+  {
+    const goto_programt::instructiont &instruction=**i_it;
+    if(instruction.code.is_not_nil())
+    {
+      for(const_depth_iteratort it=instruction.code.depth_begin();
+          it!=instruction.code.depth_end();
+          ++it)
+      {
+        if((*it).id()==ID_symbol)
+        {
+          uses.insert(*it);
```

Comment:
Would goto_rw be useful here?

Reply:
See https://github.com/diffblue/cbmc/pull/2677 for a replacement of the local_may_alias analysis used in this version with goto_rw.